lib/fluent/plugin/output.rb

```diff
@@ -1036,17 +1036,17 @@ def generate_format_proc
       #   iteration of event stream, and it should be done just once even if total event stream size
       #   is bigger than chunk_limit_size because of performance.
       def handle_stream_with_custom_format(tag, es, enqueue: false)
-        meta_and_data = {}
+        meta_and_data = Hash.new { |h, k| h[k] = [] }
         records = 0
         es.each(unpacker: Fluent::MessagePackFactory.thread_local_msgpack_unpacker) do |time, record|
           meta = metadata(tag, time, record)
-          meta_and_data[meta] ||= []
           res = format(tag, time, record)
           if res
             meta_and_data[meta] << res
             records += 1
           end
         end
+        meta_and_data.default_proc = nil
```

Comment:
> meta_and_data.default_proc = nil

Do we need this?
Looking at the following, it seems like it would work without this.

https://github.com/fluent/fluentd/blob/17d9322fadf7fad28cfa44118c7831c7a00d96ea/lib/fluent/plugin/buffer.rb#L343-L345

https://github.com/fluent/fluentd/blob/17d9322fadf7fad28cfa44118c7831c7a00d96ea/lib/fluent/plugin/buffer.rb#L653-L654

Reply:
As long as there are no incorrect references on the plug-in side, there will be no problems.
If a plugin makes an incorrect reference, it will return nil and the plugin will be aware of the problem.

To avoid worrying about the implementation of plug-ins, it pass objects that behave the same as before.

Reply:
I see.
I understood it as reverting to the default behavior so as not to have an unnecessary impact on subsequent processing.

Note: Since this hash is only written to a buffer here, we would not need to worry about processing by unspecified plugins. However, we revert to the default behavior, just in case.